controllers/logging/secret_controller.go

```diff
@@ -0,0 +1,76 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	// "sigs.k8s.io/controller-runtime/types"
+
+	loggingv1 "github.com/openshift/elasticsearch-operator/apis/logging/v1"
+)
+
+// SecretReconciler reconciles a Secret object
+type SecretReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func (r *SecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+
+	cluster := &loggingv1.Elasticsearch{}
+	esName := types.NamespacedName{
+		// TODO: how to avoid hard coding the cluster NamespaceName?
+		Namespace: "openshift-logging",
+		Name:      "elasticsearch",
+	}
+
+	err := r.Get(context.TODO(), esName, cluster)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if req.Namespace != cluster.Namespace || req.Name != cluster.Name {
+		log.Info("ES secret namespace name mismatch")
+		return ctrl.Result{}, nil
+	}
+
+	// if secret has been changed, mark cluster to be restarted
+	return ctrl.Result{}, nil
+}
+
+func esSecretUpdatePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return true
```

Comment:
we should have a function call here to check to see if there is an elasticsearch CR with the same name and same namespace as this secret... otherwise we end up updating for every secret on the cluster (which is incorrect)

Reply:
(return true if we do, false if we don't)

Reply:
Right. I missed the k8s client from the receiver `r *SecretReconciler`. So the pseudo would be

```go

esList := &ElasticsearchList{}
err := r.List(context.TODO(), esList)

for _, es := range esList {
    if e.NamespaeName == es.NamespaceName {
        return true
    }
}
return false
```

Does this look good to you? Thanks.